File: fixtures/php/src/Operations.php
<?php

declare(strict_types=1);

namespace Calculator;

final class Operations
{
    public static function add(float $a, float $b): float
    {
        return $a + $b;
    }

    public static function subtract(float $a, float $b): float
    {
        return $a - $b;
    }

    public static function multiply(float $a, float $b): float
    {
        return $a * $b;
    }

    public static function divide(float $a, float $b): float
    {
        if ($b === 0.0) {
            throw new DivisionByZeroException();
        }

        return $a / $b;
    }
}

File: fixtures/elixir/lib/operations.ex
defmodule Calculator.Operations do
  @moduledoc false

  alias Calculator.DivisionByZeroError

  @spec add(number(), number()) :: number()
  def add(a, b), do: a + b

  @spec subtract(number(), number()) :: number()
  def subtract(a, b), do: a - b

  @spec multiply(number(), number()) :: number()
  def multiply(a, b), do: a * b

  @spec divide(number(), number()) :: float()
  def divide(_a, 0), do: raise(DivisionByZeroError)
  def divide(_a, 0.0), do: raise(DivisionByZeroError)
  def divide(a, b), do: a / b
end
